### ISSUE TITLE
call to undeclared function 'lua_enablereadonlytable'; ISO C99 and later do not support implicit function declarations

### ISSUE THREAD

**Issue by terrablue** [natecraddock/lua#1]
Hey, I'm using your ziglua project to devendor lua in my redis fork. However, the vendored lua exposes two additional functions that it seems upstream doesn't otherwise expose (https://github.com/terrablue/rediz/blob/master/deps/lua/src/lua.h#L361-L362).

Would you mind exposing those functions as well? I find it very convenient to use ziglua as a dependency instead of rolling my own.

**Comment by natecraddock** [comment]
Hey! I took a look at how redis/zkv uses Lua, and it seems that they have made [several modifications](https://github.com/terrablue/zkv/blob/master/deps/README.md#lua). For example, the `readonly` attribute of tables doesn't exist in standard Lua 5.1. So this would require adding more than just the function declarations to this repo.

Even if this was only adding the following

```
LUA_API void lua_enablereadonlytable (lua_State *L, int index, int enabled);
LUA_API int lua_isreadonlytable (lua_State *L, int index);
```

I would still be hesitant to add it. Because Lua is so small and easy to modify, I'm sure Redis isn't alone in making modifications to Lua, and I don't want to set a precedent in modifying the Lua sources bundled with Ziglua. Hopefully that is understandable.

But! I do think there is a solution here. I think Ziglua should be able to support system and vendored Lua sources. Currently it only supports using the Zig package manager Lua in the build.zig.zon.

I know you want to remove your vendored Lua in zkv, but I'm really not sure you can do that because of the modifications. (You could move it to a separate repo though and use the Zig package manager to pull it in at build time.) What do you think about that?

**Comment by natecraddock** [comment]
Also I think I'll move this issue to the Ziglua repo because I think it makes more sense there

**Comment by terrablue** [comment]
Hey, thanks. So far I've got this: https://github.com/terrablue/zua, which is working.

But if possible, it would be nice to deduplicate work, and Ziglua is already doing some nice binding work. Seeing as I generally want to transition zkv to using Zig not just for the build but also for the source code, Ziglua is attractive here.

Perhaps it could be possible to use a repository as a target in addition to the targets you have in build.zig.zon? That would probably allow me to get rid my own build.zig.zon/build.zig in the zua repository.

**Comment by natecraddock** [comment]
I agree, it would be ideal if Ziglua can integrate nicely with other Lua sources. I'll take a look later today on possible solutions and follow up.

I'll move the issue
